common/src/main/java/com/wynntils/handlers/item/ItemHandler.java

```diff
@@ -148,12 +149,8 @@ private void onItemStackUpdate(ItemStack existingItem, ItemStack newItem) {
         }
     }
 
-    private CodedString getBaseName(CodedString name) {
-        int bracketIndex = name.getInternalCodedStringRepresentation().lastIndexOf('[');
-        return bracketIndex == -1
-                ? name
-                : CodedString.fromString(
-                        name.getInternalCodedStringRepresentation().substring(0, bracketIndex));
+    private StyledText getBaseName(StyledText name) {
+        return name.getPartCount() == 1 ? name : name.iterate((part, changes) -> IterationDecision.SKIP);
```

Comment:
This is most likely the most intrusive change in this PR. I tested this with consumables, and it seems to work. I think this captures the intent on the old method, in general.

Reply:
Hm. The IterationDecision SKIP means that just the first part will be kept, and the rest ignored? So it's a way of saying "give me the first part of the styled text"? We don't have a simple like `styledText.getPart(index)`? Maybe we should? And, also, the iterate would work just as well even if partcount is 1, yes? Or is that a correctness thing rather than an optimization thing?

As for the overall functionality, I guess it would make sense to break at the first format change, but I'm not 100% sure. The current solution that was looking for `[` was not ideal either. We'll probably just have to try it and see if people start reporting breakage on item overlays.

Reply:
I think we removed `getPart(index)`, it also feels a bit weird. Iterate works on any length, yes. And yes, we will see the reports, but from testing this should work.

Reply:
Maybe we could have a getFirstPart() then? It feels like a complicated code to just get the first part of a string, and it was (and still is) not really clear to me that this is really what is happening here.

Reply:
Okay. Let's do that.